src/libraries/System.Collections/src/System/Collections/Generic/Stack.cs

```diff
@@ -70,12 +70,15 @@ bool ICollection.IsSynchronized
         // Removes all Objects from the Stack.
         public void Clear()
         {
-            if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+            if (_size != 0)
```

Comment:
Same question.  All of these changes are adding a bit of code that penalizes the non-empty case to help the empty case.  Is the empty case common enough to warrant that?

Reply:
Thanks, @stephentoub. I have reverted back those unwanted changes!